### DISCUSSION TITLE
Option to use system-wide direnv

### DISCUSSION THREAD

**Discussion by glasserc** [asdf-community/asdf-direnv#58] [Q&A] [answered]
The `asdf-direnv` plugin has two functions AFAICS:

1. Install different versions of `direnv` as with any other `asdf` plugin.
2. Add a `use asdf` command to `direnv`.

These functionalities are distinct and orthogonal. Personally, I find the second functionality very valuable (thanks for the plugin!!), but I don't especially care about the first one, because:

- `direnv` is rarely a direct dependency of a project I'm working on, and the different versions are not so different that I typically care which one is being used.
- Unlike most `asdf` plugins, which manage standalone executables, `direnv` is installed via a shell hook, which means that even if you use `asdf` to install a different version of `direnv`, you might have to restart your shell in order to use it.

Because installing `direnv` via `asdf` means one more thing to update and manage, I'm inclined to just install it using my operating system and manage it there. For this reason, I'd like a way to use this plugin to only provide the `use asdf` command. I'm not sure if this is within the scope of this plugin, so if it isn't, please feel free to close :)

Right now, what happens is that any time I enter a directory with a `use asdf` is that a version of `direnv` appears on my `PATH`. I think this happens `use asdf` loads all "current" programs, which includes the "global version" of `direnv`. However, this gets executed from within `direnv`, so it may not have any effect, or at least not the intended effect.

Therefore, in order to manage `direnv` myself, I'd like to not have to set a global version of `direnv`. There are at least these two things that rely on this global version so far: 

- Finding the hook `command-hook-asdf.bash` relies on having a `direnv` version selected. This seems unfortunate since this file comes from the `asdf-direnv` plugin and not `direnv` itself.
- The related `command.bash` script relies on `asdf` to execute `direnv`. It would be nice if it could optionally use whatever was already in the `PATH`.

**Comment by vic** [comment, marked as answer]
After installing this plugin you let asdf know that you want to use the `direnv` binary installed by your system
just by having a `.tool-versions` file like this:

```
# this is my global ~/.tool-versions file
direnv system
```

and then just create a directory for `asdf-direnv` to store cached environments.

```
$ mkdir -p ~/.asdf/installs/direnv/system/env
```

If I recall correctly (at least when I implemented asdf external commands) they don't depend on any version to be installed, since they are part of the plugin itself. And as long as you have the plugin installed, the extension commands like `command-hook-asdf.bash` should be found by asdf.

**Reply by andrewcrook** [reply]
Can you help please.
This isn’t working for me but it is set up like yours. https://github.com/asdf-community/asdf-direnv/issues/83
Any ideas?

**Reply by michi-zuri** [reply]
please see my answer in #83

**Comment by glasserc** [comment]
Ah, I didn't know about `system`, that's super cool, thanks! I was able to follow your instructions and once I did the `mkdir`, everything worked perfectly. Is there maybe a way for `asdf-direnv` to do the `mkdir` on the user's behalf? (Until I did, I got some very puzzling error messages: `direnv: referenced  does not exist`.) Otherwise I think this can be closed. Thanks again!

**Reply by comfortablynick** [reply]
This got me as well. It seems like asdf-direnv could make this directory when “system” is being used. I had to install direnv via homebrew for now on my M1 Mac.
